woocommerce/woocommerce-gutenberg-products-block
# Product gallery - fix “View all” link

## What

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11988

This PR adds specific styling to the "View All" link and thumbnails count depending on how many thumbnails are displayed.

- When set to 3 - 5 thumbnails both thumbnail count and "View All" link is shown.
- When set to 6 - 8 thumbnails only count is shown because "View All" link doesn't have enough space to display nicely.

## Why



This is so that shoppers will have a better visual experience and not have the "View All" link out of place or not showing completely.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Appearance->Editor->Templates->Single Product.
2. Add the Product Gallery block.
3. Go to a product in the frontend with 8 gallery images added.
4. Ensure you see both the "Count" and "View All" link and are displayed nicely.
5. Go back to the Editor Product Gallery block and change to show 8 thumbnails.
6. Go back to the frontend product with 8 gallery images.
7. Ensure you see only the "Count" being displayed and that it is showing nicely and centered.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|![file.png](https://github.com/woocommerce/woocommerce-blocks/assets/2132595/5d61b08d-2ae1-481b-8e33-d298eb61584a)        |![file.png](https://github.com/woocommerce/woocommerce-blocks/assets/2132595/64fc45cb-2a48-4b84-9115-2c19cc0bd837)       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
